include/bluetooth/mesh/gen_battery_cli.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file
+ * @defgroup bt_mesh_gen_battery_cli Generic Battery Client model
+ * @{
+ * @brief API for the Generic Battery Client model.
+ */
+
+#ifndef BT_MESH_GEN_BATTERY_CLI_H__
+#define BT_MESH_GEN_BATTERY_CLI_H__
+
+#include <bluetooth/mesh/gen_battery.h>
+#include <bluetooth/mesh/model_types.h>
+
+struct gen_battery_cli;
+
+/** @def GEN_BATTERY_CLI
+ *
+ * @brief Initialization parameters for a @ref gen_battery_cli instance.
+ *
+ * @param[in] _status_handler Optional status message handler.
+ */
+#define GEN_BATTERY_CLI(_status_handler)                                       \
+	{                                                                      \
+		.status_handler = _status_handler,                             \
+		.pub = {.msg = NET_BUF_SIMPLE(2 + 0 + 4) }                     \
```

Comment:
IMO, the `0` representing the payload here doesn't really need its own define, but it doesn't really hurt, I suppose. I have [a PR](https://github.com/zephyrproject-rtos/zephyr/pull/19676) pending in upstream Zephyr for fixing the 2 and the 4 globally.

Reply:
Note that the model publication message does *not* require space for the MIC at the end. This is because it gets copied to a stack-internal buffer, e.g. to handle publish retransmissions.

Reply:
To follow up on my previous comment - we might want to introduce a convenience macro for defining such messages, e.g. with a _PUB suffix

Reply:
It should be possible for a model to publish/send a message with either 4 byte or 8 byte MIC. The specification leaves this choice to the users. Will it be good to add such flexibility as a separate task?